packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/DeleteCategoriesTests.java

```diff
@@ -0,0 +1,133 @@
+        user = dataUser.createRandomTestUser();
+    }
+
+    /**
+     * Check we can delete a category.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteCategory()
+    {
+        STEP("Create a category and send a request to delete it.");
+        RestCategoryModel aCategory = createCategory();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingCategory(aCategory).deleteCategory();
+        restClient.assertStatusCodeIs(NO_CONTENT);
+
+        STEP("Ensure that the category has been deleted by sending a GET request and receiving 404.");
+        restClient.authenticateUser(user).withCoreAPI().usingCategory(aCategory).getCategory();
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Check we get an error when trying to delete a category as a non-admin user.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteCategoryAsRegularUser_andFail()
+    {
+        RestCategoryModel aCategory = createCategory();
+        restClient.authenticateUser(user).withCoreAPI().usingCategory(aCategory).deleteCategory();
+        restClient.assertStatusCodeIs(FORBIDDEN).assertLastError().containsSummary("Current user does not have permission to delete a category");
+    }
+
+    /**
+     * Check we receive 404 error when trying to delete a category with a non-existent node id.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteNonExistentCategory()
+    {
+        STEP("Get category with non-existent id");
+        final RestCategoryModel rootCategory = new RestCategoryModel();
+        final String id = "non-existing-dummy-id";
+        rootCategory.setId(id);
+
+        STEP("Attempt to delete category with non-existent id and receive 404");
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingCategory(rootCategory).deleteCategory();
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Attempt to delete a category when providing a node id that doesn't belong to a category
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteCategory_givenNonCategoryNodeId()
+    {
+        STEP("Create a site and a folder inside it");
+        final SiteModel site = dataSite.usingUser(user).createPublicRandomSite();
+        final FolderModel folder = dataContent.usingUser(user).usingSite(site).createFolder();
+        String id = folder.getNodeRef();
+
+        STEP("Create a category, set its id to the folder id and attempt to delete it");
+        final RestCategoryModel aCategory = new RestCategoryModel();
+        aCategory.setId(id);
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingCategory(aCategory).deleteCategory();
+        restClient.assertStatusCodeIs(BAD_REQUEST).assertLastError().containsSummary("Node id does not refer to a valid category");
+    }
+
+    public RestCategoryModel createCategory()
```

Comment:
I think Krys introduced some common utility class to handle this and probably some more. 
Should we wait for his PR to be merged or we'll handle making use of common code later?

Reply:
Krystian suggested that I merge this PR and he would update parts of this code when he does merge his PR and the helper methods are available. Is that okay?

Reply:
Yep - that's great @gevagel . Go enjoy your holiday! :christmas_tree:

Reply:
Thank you all! 🎄 🎅

Reply:
@gevagel you can merge it without worries. I will adjust the common code and use it from your code.